quodlibet/tests/test_browsers_playlists.py

```diff
@@ -262,13 +262,21 @@ def test_default_display_pattern(self):
         self.failUnless("<~name>" in pattern_text)
 
     def test_drag_data_get(self):
+        # Temporarily replace code that creates a prompt for the user
```

Comment:
So this is exactly what you have to do (monkey-patching and mutating global state) if you don't have the loose coupling.

If we'd injected the confirmer function / object (e.g. as an optional constructor parameter to the `PlaylistsBrowser`), we could simply instantiate the browser here in tests passing in a _mock_ confirmer - one with a `run` implementation similar to the one you've patched in below - but avoiding any monkey patching.